src/protocol/pubsub0/sub.c

```diff
@@ -414,7 +414,7 @@ sub0_recv_cb(void *arg)
 	if (msg != NULL) {
 		nni_msg_free(msg);
 	}
-
+	nni_pollable_raise(sock->recvable);
```

Comment:
This isn't quite right, but it's close.  The issue here is that we don't want to raise this FD unless we passed the SUB filter.  So what needs to happen is up around line 382 we need to set a flag about whether this was matched or not, and then only do the raise down here if we matched.  Otherwise we may wind up doing a lot more spurious wakeups.